s4/s4.py

```diff
@@ -661,7 +669,7 @@ class StackedModel(nn.Module):
     decode: bool = False
 
     def setup(self):
-        self.encoder = nn.Dense(self.d_model)
+        self.encoder = nn.Embed(num_embeddings=self.n_embed, features=self.d_model)
```

Comment:
have to change this part of S4 in order to run BN

Reply:
I think this needs to be gated to not break the other code.

Reply:
got it, we will just change the dataset.py for now not to touch the model itself.